pkg/cluster/defaults.go

```diff
@@ -88,3 +107,34 @@ func setMachineHealthCheckTimeoutDefaults(cluster *anywherev1.Cluster, nodeStart
 		}
 	}
 }
+
+// setMachineHealthCheckMaxUnhealthyDefaults sets default maxUnhealthy values for cluster's machine health checks.
+func setMachineHealthCheckMaxUnhealthyDefaults(cluster *anywherev1.Cluster, globalMaxUnhealthy, workerMaxUnhealthy intstr.IntOrString) {
+	topLevelMaxUnhealthyUndefined := true
+	if cluster.Spec.MachineHealthCheck.MaxUnhealthy == nil {
+		cluster.Spec.MachineHealthCheck.MaxUnhealthy = &globalMaxUnhealthy
+	} else {
+		topLevelMaxUnhealthyUndefined = false
+	}
+
+	if cluster.Spec.ControlPlaneConfiguration.MachineHealthCheck == nil {
+		cluster.Spec.ControlPlaneConfiguration.MachineHealthCheck = &anywherev1.MachineHealthCheck{}
+	}
+
+	if cluster.Spec.ControlPlaneConfiguration.MachineHealthCheck.MaxUnhealthy == nil {
+		cluster.Spec.ControlPlaneConfiguration.MachineHealthCheck.MaxUnhealthy = &globalMaxUnhealthy
+	}
+
+	for i := range cluster.Spec.WorkerNodeGroupConfigurations {
```

Comment:
That's what I had before but the cluster object wasn't taking the values. Then I realized, since we are creating a copy of the `wng` object, the original cluster spec doesn't get affected. So we need to range over and directly affect the cluster.Spec object instead of the `wng` object
Similar to this fix - https://github.com/aws/eks-anywhere/commit/ae743af72bce0434112742f904967fa557f480a7